### ISSUE TITLE
Validation: Strip blacklisted a[rel] values

### ISSUE THREAD

**Issue by kraftbj** [Automattic/amp-wp#237]
<strike>`rel` values on `a` tags must be part of the HTML spec. We should remove nonstandard tags.

https://www.w3.org/TR/html5/links.html#linkTypes

I'm unsure if other specs extend this. Seen specifically with `rel="homepage"` (added on my site via an old tool to insert links to related content, etc.</strike>

Looks like the AMP validator switched things around to only flag empty and blacklisted values. We should strip those out and ignore the rest.

https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L1015

**Comment by mjangda** [comment]
I believe that this is the list the validator goes by: http://microformats.org/wiki/existing-rel-values

**Comment by mjangda** [comment]
Looks like the AMP validator switched things around to only flag empty and blacklisted values

https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L1015

**Comment by kienstra** [comment]
**Looks To Be Resolved**

There's now a [regex](https://github.com/Automattic/amp-wp/blame/develop/includes/sanitizers/class-amp-allowed-tags-generated.php#L50) to check for the allowed 'rel' values for `<a>` elements. This is consistent with the AMP specification's [blacklisted_value_regex](https://github.com/ampproject/amphtml/blame/master/validator/validator-main.protoascii#L1192). But this looks to have been fixed before the work for #804.

```
'rel' => array(
	'blacklisted_value_regex' => '(^|\\s)(components|dns-prefetch|import|manifest|preconnect|prefetch|preload|prerender|serviceworker|stylesheet|subresource|)(\\s|$)',
),
```

For example, this element:
```
<a href="http://example.com/baz" rel="components">Example Text</a>
```
will have its disallowed `rel` value stripped:
```
<a href="http://example.com/baz" rel="components">Example Text</a>
```
